web/components/profile/general.jsx

```diff
@@ -1,12 +1,199 @@
+import AlertActions from '../../actions/alert-actions';
+import DatePicker from '../controls/date-picker.jsx';
+import Formsy from 'formsy-react';
+import PropTypes from 'prop-types';
 import React from 'react';
+import request from '../../request-agent';
+import SelectBox from '../controls/select-box.jsx';
+import Spinner from '../controls/spinner.jsx';
+import TextBox from '../controls/text-box.jsx';
+import UserActions from '../../actions/user-actions';
+import UserStore from '../../stores/user-store';
+
+import {
+	Button
+} from 'react-bootstrap';
+
+const ALERT_KEY = 'profile';
 
 class GeneralInfo extends React.Component {
+	constructor() {
+		super();
+		this.state = {
+			currentUser: UserStore.getState().currentUser,
+			isSaving: false
+		};
+		this.onUserChanged = this.onUserChanged.bind(this);
+		this.submit = this.submit.bind(this);
+	}
+
+	componentDidMount() {
+		UserStore.listen(this.onUserChanged);
+		AlertActions.dismissAlert(ALERT_KEY);
+	}
+
+	componentWillUnmount() {
+		UserStore.unlisten(this.onUserChanged);
+	}
+
+	onUserChanged() {
+		this.setState({
+			currentUser: UserStore.getState().currentUser,
+			isSaving: this.state.isSaving
+		});
+	}
+
+	submit(model, reset, invalidate) {
+		AlertActions.dismissAlert(ALERT_KEY);
+		this.setState({
+			currentUser: this.state.currentUser,
+			isSaving: true
+		});
+
+		model.numberOfDives = model.numberOfDives || null;
+		model.diverType = model.diverType || null;
+
+		request
+			.put(`/api/users/${this.props.match.params.userName}/`)
+			.send(model)
+			.then(res => {
+				if (this.state.currentUser.userName === this.props.match.params.userName) {
+					UserActions.updateProfile(res.body);
+				}
+
+				AlertActions.showSuccess(
+					ALERT_KEY,
+					'Profile Saved',
+					'Your profile information has been saved successfully.');
+			})
+			.catch(res => {
+				if (res.status === 403) {
+					// Attempt to change e-mail to one that already exists in the system.
+					return invalidate({
+						email: 'Unable to change e-mail address. This address is already taken!'
+					});
+				}
+
+				AlertActions.handleErrorResponse(ALERT_KEY, res);
+			})
+			.finally(() => {
+				this.setState({
+					currentUser: this.state.currentUser,
+					isSaving: false
+				});
+			});
+	}
+
 	render() {
+		const user = this.state.currentUser;
+
 		return (
 			<div>
 				<h3>My Info</h3>
+				<p className="help-block">
+					<strong>Note:</strong> Your profile information is not shared with or sold to anyone not directly affiliated with this site.
```

Comment:
Consider rewording this to say that profile information is not shared _at all_ and mention that profile info is `private` by default.